### PR TITLE
Restrict output metadata value type

### PR DESCRIPTION
**Pull request by mattdean-digicatapult**
Allows restricting the type of a metadata value for a specific output key on a specific output index

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -192,6 +196,11 @@ In order to use the API within `polkadot.js` you'll need to configure the follow
     "metadata_key": "TokenMetadataKey",
     "metadata_value": "TokenMetadataValue"
   },
+  "FixedOutputMetadataValueType": {
```
Comment (jonmattgray) [inline README.md:199]: should have restriction on the end? `FixedOutputMetadataValueTypeRestriction`

Comment (mattdean-digicatapult) [reply README.md]: renamed

File: pallets/process-validation/src/restrictions.rs [hunk 2]
```diff
@@ -44,22 +45,28 @@ where
         metadata_key: TokenMetadataKey,
         metadata_value: TokenMetadataValue,
     },
+    FixedOutputMetadataValueType {
+        index: u32,
+        metadata_key: TokenMetadataKey,
+        metadata_value_type: TokenMetadataValueDiscriminator,
```
Comment (jonmattgray) [inline pallets/process-validation/src/restrictions.rs:51]: does `TokenMetadataValueDiscriminator` end up as dropdown of types in the polkadot explorer?

Comment (mattdean-digicatapult) [reply pallets/process-validation/src/restrictions.rs:51]: I don't believe so. I believe the frontend uses the supplied types to generate names. These names are all internal to the build

**Review by jonmattgray — APPROVED** [review]

**Pull request merged**